scripts/cre_sync/docs/implementation.md

```diff
@@ -0,0 +1,275 @@
+CRE Utilities Implementation
+===
+
+Introduction
+---
+
+In order to implement the CRE v0.91 design defined [here|] and to ensure the [requirements|"./requirements.md"] are satisfied several utilities need to be created.
+This application needs to be easily deployable and also define a feature-complete library which would allow for others to adopt the CRE structure.
+
+Suggested Modifications:
+
+* Versioning: For CRE to be resilient, the files need to have an identifier that allows for the format to change without breaking compatibility, it is therefore recommended for files to include the Version of the CRE parser they are targetting. The parser can then parse the fields it knows about for the versions it is aware off.
+
+* Interoperability: the fiels need to be parseable easily by widely used libraries, while the example files listed in the requirements document are valid YAML, a slightly modified format is suggested. Please refer to the section "File Format" for rationale and more information.
+
+* Mapping files: Allowing standards writers to submit a file where they map to a specific CRE is a good idea, it would be good to expand this by allowing standards writers to link to other standards as well, within the same mapping file.
+This abstracts mapping files away from a single standard and instead allows standards writers to provide a document hierarchy that contains CREs and/or other standards.
+
+* The url for the refer use-case is of the form: `http://cre.com/refer?273-300` it is suggested to add a variable and thus transform it to:
+`http://cre.com/refer?cre=273-300` this would enable the addition of other variables for future use cases, e.g.:
+`http://cre.com/refer?cre=273-300&link=nist&link=asvs&link=top10-2017`
+
+* Higher level concepts: TODO(Spyros): merge cregroups and cres, there is not point in making them different
+
+* CRE Version Pinning, the TOP 10 project has asked for links to CREs as they were at a point in time. I don't have a good solution to this for mapping files, it could be solved by linking to a git sha but then I'm not sure how this would work for the frontend, an alternative would be to make every CRE, Standard and Link versioned and link against a version, but then they loose all the new mappings.
```

Comment:
then how would this be implemented in the backned?

Reply:
Hmmm, I'd say this will be tied to the data that the backend will load.
I need to research this a bit. Pretty sure there's a nice way to do this.
I believe we're gonna have to rely on branches, and not tags, if we want to keep supporting the versions and updating them.
@kingthorin any ideas on this, or anyone that is more in-depth with this matter?